tests/suites/hup/tests/smoke.js

```diff
@@ -118,5 +133,20 @@ module.exports = {
 			'0',
 			'Volume should not be lost during HUP',
 		);
+
+		test.comment(`Checking for under-voltage reports in kernel logs after hup...`);
+		result = await this.context
+				.get()
+				.worker.executeCommandInHostOS(
+					`dmesg | grep -q "Under-voltage detected" ; echo $?`,
+					this.context.get().link,
+				);
+
+		if (result.includes('0')) {
+			test.comment(`not ok! - Under-voltage detected on device, please check power source and cable!`);
+		} else {
+			test.comment(`ok - No under-voltage reports in the kernel logs`);
+		}
+
```

Comment:
Do you only want these checks in the smoke test, or the other hup tests as well?

It might be worth exporting it as a function in `hup/suite.js` so we can avoid duplicating the code.

Reply:
Sorry Kyle. I misread your question. It's actually only meant for OS and for the HUP smoke test. The smoke test is the first one that's run, so if the problem is present there, there's no need to check in the subsequent ones.